### PR TITLE
New RBAC UI

### PR DESCRIPTION
**Pull request by aptkingston**
## Description

image

This PR adds a new visual way of defining roles for your apps. Custom roles can be created, edited and deleted using this new RBAC editor, and inheritance can be defined using intuitive drag-and-drop connections. Inheriting multiple roles is now possible too.

Clarification of functionality:
- Power user role remains for existing apps, but will not be created in new apps
- Power user role cannot be edited and cannot inherit custom roles, but custom roles may inherit it
- Fully backwards compatible with existing custom roles structures
- Full multiplayer support for custom role CRUD
- Positions of roles are not currently saved, by design. The layout will reset to auto-layout when refreshing the page.

## Feature branch env
[Feature Branch Link](http://fb-new-rbac-ui.fb.qa.budibase.net)

### CODE REVIEW

**Comment by PClmnt** [comment]
Just wanna say, that UI is absolutely beautiful!

File: packages/server/src/api/controllers/role.ts [hunk 1]
```diff
@@ -177,6 +178,7 @@ export async function save(ctx: UserCtx<SaveRoleRequest, SaveRoleResponse>) {
       },
     })
   }
+  builderSocket?.emitRoleUpdate(ctx, role)
```
Comment (aptkingston) [inline packages/server/src/api/controllers/role.ts:181]: The only backend change is multiplayer support for role changes.

File: packages/client/src/components/devtools/DevToolsHeader.svelte [hunk 2]
```diff
@@ -31,12 +32,12 @@
...
     // make sure correct before starting
     await devToolsStore.actions.changeRole(SELF_ROLE)
-    staticRoleList = await roleStore.actions.fetchAccessibleRoles()
+    roles = await API.getRoles()
   })
 </script>
 
 <div class="dev-preview-header" class:mobile={$context.device.mobile}>
-  <Heading size="XS">Preview</Heading>
+  <Heading size="XS">Previ werewr eww</Heading>
```
Comment (deanhannigan) [inline packages/client/src/components/devtools/DevToolsHeader.svelte:40]: 🥚

**Review by deanhannigan — APPROVED** [review]
LGTM! Any issues found during testing are now resolved 🎉 . Fantastic work @aptkingston. The new role UX is amazing!

File: packages/client/src/components/devtools/DevToolsHeader.svelte (hunk 2 above)
Comment (aptkingston) [reply packages/client/src/components/devtools/DevToolsHeader.svelte:40]: Thank you 😂  May have been testing things...

**Review by mike12345567 — APPROVED** [review]

**Pull request merged**